### PR TITLE
feat: migrate handler to modelpb and prepare for decoder migration

### PR DESCRIPTION
**Pull request by kruskall**
Preparation for migrating the decoder to protobuf:

- update handlers to protobuf models
- update elasticapm code to protobuf models (needed to remove the protobatchprocessor compatibility layer from APM Server)

One of the test is relying on reflection and it's doesn't work well with protobuf so I've skipped it for now.

Related to https://github.com/elastic/apm-data/issues/52
Blocked by https://github.com/elastic/apm-data/pull/60

### CODE REVIEW

File: input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go [hunk 1]
```diff
@@ -263,6 +265,10 @@ func SetZeroStructValue(i interface{}, callback func(string)) {
 // that values are equal to expected values
 func AssertStructValues(t *testing.T, i interface{}, isException func(string) bool,
 	values *Values) {
+	if true {
+		// TODO FIX no op protobuf support
+		return
```
Comment (axw) [inline input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:270]: I'm not particularly comfortable with this. If we're merging without these tests, please open an issue so we can get it back in ASAP. Perhaps we should consider getting rid of the reflection approach.

File: input/elasticapm/internal/modeldecoder/rumv3/decoder.go [hunk 2]
```diff
@@ -146,12 +153,14 @@ func DecodeNestedTransaction(d decoder.Decoder, input *modeldecoder.Input, batch
...
-		*batch = append(*batch, event)
+		var out modelpb.APMEvent
+		event.ToModelProtobuf(&out)
+		*batch = append(*batch, &out)
 	}
 	spans := (*batch)[offset:]
 	for i, s := range root.Transaction.Spans {
 		if s.ParentIndex.IsSet() && s.ParentIndex.Val >= 0 && s.ParentIndex.Val < len(spans) {
-			spans[i].Parent.ID = spans[s.ParentIndex.Val].Span.ID
+			spans[i].Parent.Id = spans[s.ParentIndex.Val].Span.Id
```
Comment (axw) [inline input/elasticapm/internal/modeldecoder/rumv3/decoder.go:163]: Are `Parent` and `Span` guaranteed to be non-nil, or should we be checking them here?

**Review by axw — APPROVED** [review]

File: input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go (hunk 1 above)
Comment (axw) [reply input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go:270]: I removed the TODO and updated tests to remove/replace the AssertStructValues usage where I thought it made sense: https://github.com/axw/apm-data/commit/e3a387cc21fad079d6bf170164dcd8c9255e6d12. Please have a look and feel free to cherry-pick that if you think it looks sense.

Comment (kruskall) [reply input/elasticapm/internal/modeldecoder/modeldecodertest/populator.go]: Thanks for this! I cherry-picked your commit!

File: input/elasticapm/internal/modeldecoder/rumv3/decoder.go (hunk 2 above)
Comment (kruskall) [reply input/elasticapm/internal/modeldecoder/rumv3/decoder.go]: Updated!

**Comment by kruskall** [comment]
signing requirements :smiling_face_with_tear:

**Pull request merged**